integrations/server_test.go

```diff
@@ -0,0 +1,519 @@
+package integrations
+
+// This was intentionally added under the integrations package and not the integrations test package
+// so that changes in other parts of the code base that may have an effect on these test will not
+// compile until they are fixed.
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/chronograf"
+	"github.com/influxdata/chronograf/bolt"
+	"github.com/influxdata/chronograf/oauth2"
+	"github.com/influxdata/chronograf/server"
+)
+
+func TestServer(t *testing.T) {
+	type fields struct {
+		Organizations []chronograf.Organization
+		Users         []chronograf.User
+		Sources       []chronograf.Source
+		Servers       []chronograf.Server
+		Layouts       []chronograf.Layout
+		Dashboards    []chronograf.Dashboard
+	}
+	type args struct {
+		server    *server.Server
+		method    string
+		path      string
+		payload   interface{} // Expects this to be a json serializable struct
+		principal oauth2.Principal
+	}
+	type wants struct {
+		statusCode  int
+		contentType string
+		body        string
+	}
+
+	tests := []struct {
+		name    string
+		subName string
+		fields  fields
+		args    args
+		wants   wants
+	}{
+		{
+			name:    "GET /users",
+			subName: "User Not Found in the Default Organization",
+			fields:  fields{},
+			args: args{
+				server: &server.Server{
+					GithubClientID:     "not empty",
+					GithubClientSecret: "not empty",
+				},
+				method: "GET",
+				path:   "/chronograf/v1/users",
+				principal: oauth2.Principal{
+					Organization: "0",
+					Subject:      "billibob",
+					Issuer:       "github",
+					ExpiresAt:    time.Now().Add(10 * time.Second),
```

Comment:
will this be sufficient in case there are ever bottlenecks in the test, for some unknown reason?

Reply:
No single one of these tests should take more than 10s. If they do theres something extra-ordinarily wrong. 

Just to be cautious though, we can move this logic into the test body.